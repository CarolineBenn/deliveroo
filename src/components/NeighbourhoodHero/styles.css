.neighbourhood__hero-container {
  background-size: cover;
  background-repeat: no-repeat;
  background-position: bottom center;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 50vh;
  padding: 1rem;
  color: white;
}

.neighbourhood__hero-details { 
  position: relative;
  text-align: center;
  z-index: 1000;
}
