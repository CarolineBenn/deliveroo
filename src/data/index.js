export const restaurants = [
  {
    id: 1,
    name: 'Bobo Social',
    tags: ['Vietnamese'],
    priceCategory: '££',
    time: '30',
    permalink: '/bobo-social'
  },
  {
    id: 2,
    name: 'Tsunami',
    tags: ['Vietnamese'],
    priceCategory: '££',
    time: '30',
    permalink: 'tsunami'
  },
  {
    id: 3,
    name: 'Wild Game Co',
    tags: ['Vietnamese'],
    priceCategory: '££',
    time: '30',
    permalink: 'wild-game-co'
  }
];

export const restaurant = {
  name: 'Bobo Social',
  tags: ['Burgers'],
  priceCategory: '££',
  blurb: 'Serving the highest-quality burgers and sides with an array of creative and tasty toppings, Bobo Social cooks its rare-breed beed in a special oven to take you to burger heaven!',
  menu: [
    {
      category: 'Burgers',
      options: [
        {
          id: 1,
          name: 'Bobo',
          desc: '6oz rare-breed beef, lettuce, tomatoes, charcoal-roasted red onions, BOBO sauce',
          price: '£8.95',
        },
        {
          id: 2,
          name: 'Bobo with Cheese',
          desc: '6oz rare-breed beef, Westcombe Cheddar, lettuce, tomatoes, charcoal-roasted red onions, BOBO sauce',
          price: '£9.95',
        }
      ],  
    }
  ],
};


export const neighbourhood = {
  name: 'Hampstead',
  description: 'Hampstead is a great place to while away a whole day. As well as three of London’s biggest and best museums, there’re plenty of other things to do that are sure to entertain and amuse everyone in your group.',
  count: 34,
  restaurants: restaurants,
};
