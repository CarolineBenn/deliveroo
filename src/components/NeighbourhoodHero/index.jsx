import React from 'react';

import './styles.css';

const NeighbourhoodHero = ({ count, description, hero, name }) => (
  <div className="neighbourhood__hero-container" style={{ backgroundImage: `url(${hero})` }}>
    <div className="neighbourhood__hero-details">
      <span>{count} restaurants delivering to</span>
      <h1 className="neighbourhood_hero-name">{name}</h1>
      <p>{description}</p>
    </div>
    <div className="overlay" />
  </div>
);

export default NeighbourhoodHero;
