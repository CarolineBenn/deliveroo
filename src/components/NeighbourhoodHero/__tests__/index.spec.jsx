import React from 'react';
import renderer from 'react-test-renderer';

import { neighbourhood } from '../../../data';

import NeighbourhoodHero from '..';

test('NeighbourhoodHero renders correctly', () => {
  const tree = renderer
    .create(
      <NeighbourhoodHero
        name={neighbourhood.name}
        description={neighbourhood.description}
        count={neighbourhood.count}
        hero={neighbourhood.hero}
      />,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});