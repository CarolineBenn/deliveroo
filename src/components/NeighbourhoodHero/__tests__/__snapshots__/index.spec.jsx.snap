// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeighbourhoodHero renders correctly 1`] = `
<div
  className="neighbourhood__hero-container"
  style={
    Object {
      "backgroundImage": "url(/images/neighbourhoodhero.png)",
    }
  }
>
  <div
    className="neighbourhood__hero-details"
  >
    <span>
      34
       restaurants delivering to
    </span>
    <h1>
      Hampstead
    </h1>
    <p>
      Hampstead is a great place to while away a whole day. As well as three of London’s biggest and best museums, there’re plenty of other things to do that are sure to entertain and amuse everyone in your group.
    </p>
  </div>
  <div
    className="overlay"
  />
</div>
`;
